locations/spiders/petrol_bg.py

```diff
@@ -12,4 +13,18 @@ def parse_item(self, item, location):
         if m := re.match(r"^(\d+) (.+)$", item["name"]):
             item["ref"] = m.group(1)
             item["name"] = m.group(2)
+
+        categories = location["categories"].split(",")
```

Comment:
```suggestion
        categories = location.get("categories", "").split(",")
```

Sounds like a few don't have any categories.